utils/filesystem/embedfs_test.go

```diff
@@ -0,0 +1,155 @@
+package filesystem
+
+import (
+	"embed"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"testing"
+
+	"github.com/bxcodec/faker/v3"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/ARM-software/golang-utils/utils/commonerrors"
+	"github.com/ARM-software/golang-utils/utils/commonerrors/errortest"
+)
+
+//go:embed *
```

Comment:
should this embed the files we are testing. I don't know how `*` will behave?

Reply:
this embeds all files in current directory only when building the test binaries